From 249a880d5e0ace37a688c6959fd6851b3765794e Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Fri, 21 Sep 2012 18:57:23 -0300
Subject: [RHEL6 qemu-kvm PATCH 02/23] usb-ehci: drop unused isoch_pause
 variable

RH-Author: Hans de Goede <hdegoede@redhat.com>
Message-id: <1348253864-3050-2-git-send-email-hdegoede@redhat.com>
Patchwork-id: 42177
O-Subject: [RHEL-6.4 qemu-kvm PATCH 01/22] usb-ehci: drop unused isoch_pause variable
Bugzilla: 805172
RH-Acked-by: Uri Lublin <uril@redhat.com>
RH-Acked-by: Arnon Gilboa <agilboa@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Upstream commit: fcb70eca3f14349a0ef8e74dc808360f81592026
Conflicts: hw/usb-ehci.c
---
 hw/usb-ehci.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/usb-ehci.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/hw/usb-ehci.c b/hw/usb-ehci.c
index 2e82833..875b526 100644
--- a/hw/usb-ehci.c
+++ b/hw/usb-ehci.c
@@ -418,7 +418,6 @@ struct EHCIState {
 
     USBPacket ipacket;
     uint8_t ibuffer[BUFF_SIZE];
-    int isoch_pause;
 
     uint32_t last_run_usec;
     uint32_t frame_end_usec;
@@ -887,7 +886,6 @@ static void ehci_reset(void *opaque)
 
     s->astate = EST_INACTIVE;
     s->pstate = EST_INACTIVE;
-    s->isoch_pause = -1;
     s->attach_poll_counter = 0;
 
     for(i = 0; i < NB_PORTS; i++) {
@@ -2143,9 +2141,7 @@ static void ehci_frame_timer(void *opaque)
 
     for (i = 0; i < frames; i++) {
         if ( !(ehci->usbsts & USBSTS_HALT)) {
-            if (ehci->isoch_pause <= 0) {
-                ehci->frindex += 8;
-            }
+            ehci->frindex += 8;
 
             if (ehci->frindex == 0x00002000) {
                 ehci_set_interrupt(ehci, USBSTS_FLR);
-- 
1.7.11.4

